### PR TITLE
#670: Added optional local-config file

### PR DESCRIPTION
**Pull request by gdasbach**
Added additional property for local-config, where developer can set their local configuration property file to override settings in the repo config.properties file

Can be set at launch with -Dlocal-config=etc/local.properties

### CODE REVIEW

**Comment by gdasbach** [comment]
Looks like this is exactly the same CI problem seen on #669.  Rebuild fixed that check, so I'll try the same fix again.

File: configuration/src/main/java/com/proofpoint/configuration/ConfigurationFactoryBuilder.java [hunk 1]
```diff
@@ -20,10 +20,7 @@
 import com.proofpoint.configuration.Problems.Monitor;
 
 import javax.annotation.Nullable;
-import java.io.FileInputStream;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.Reader;
+import java.io.*;
```
Comment (johngmyers) [inline configuration/src/main/java/com/proofpoint/configuration/ConfigurationFactoryBuilder.java]: Import the Proofpoint code style into your IDE. Wildcard imports are not permitted.

Comment (gdasbach) [reply configuration/src/main/java/com/proofpoint/configuration/ConfigurationFactoryBuilder.java]: Fixed.  To answer the other question, in a large team, the project config.properties file is edited quite frequently with new settings for feature implemented by developers.  It's therefore fairly common for a merge with upstream to change values in that file.  If we're using a separate config.properties, we don't see those changes unless we manually port the changes into our local copy.  If we edit the config.properties to contain our local overrides, we frequently have merge problems.  This solution allows us to maintain local overrides in a separate file while still reading configuration from the default config.properties file.

**Comment by gdasbach** [comment]
Left this comment on the previous change request, but haven't heard anything in response, so copying to this discussion.

> To answer the other question, in a large team, the project config.properties file is edited quite frequently with new settings for features implemented by developers. It's therefore fairly common for a merge with upstream to change values in that file. If we're using a separate config.properties, we don't see those changes unless we manually port the changes into our local copy. If we edit the config.properties to contain our local overrides, we frequently have merge problems. This solution allows us to maintain local overrides in a separate file while still reading configuration from the default config.properties file.

**Comment by johngmyers** [comment]
Closing for now per offline conversation.

**Pull request closed without merging**